inbm-lib/inbm_common_lib/utility.py

```diff
@@ -205,7 +205,9 @@ def is_within_directory(directory: str, target: str) -> bool:
     return prefix == abs_directory
 
 
-def safe_extract(tarball: tarfile.TarFile, path: str = ".", members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
+def safe_extract(tarball: tarfile.TarFile,
+                 path: str = ".",
+                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
```

Comment:
**Validation**
**Suggestion:** Validate the 'path' parameter to ensure it is a directory before proceeding with the extraction.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def safe_extract(tarball: tarfile.TarFile,
                 path: str = ".",
                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
    if not os.path.isdir(path):
        raise ValueError(f"The path {path} is not a valid directory.")
```